engines/mohawk/myst_state.cpp

```diff
@@ -227,6 +234,8 @@ void MystGameState::updateMetadateForSaving(const Common::String &desc) {
 	_metadata.saveMinute = t.tm_min;
 	_metadata.saveDescription = desc;
 	_metadata.totalPlayTime = _vm->getTotalPlayTime();
+	_metadata.version = 2;
+	_metadata.autoSave = (uint8) Mohawk::kAutoSaveAllowed && (slot == Mohawk::kAutoSaveSlot);
```

Comment:
You shouldn't use a global variable for knowing if this an autosave. It adds tight coupling to things that don't need to know much about each other.. There could be parameter `bool isAutosave` in this method.

Reply:
I don't think it's correct to decide whether autosave is allowed in querySaveMetaInfos. The game could be started without it being called. It's probably a lot easier and safer to have a method isAutoSaveAllowed that is called when trying to autosave that opens the save in slot 0 if there is one and checks if it's an autosave.